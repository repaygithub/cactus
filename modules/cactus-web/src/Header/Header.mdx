---
name: Header
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import Header from './Header'
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live'
import { livePreviewStyle } from '../helpers/constants'
import Button from '../Button/Button'
import Breadcrumb from '../Breadcrumb/Breadcrumb'

# Header

The `Header` component allows you to set a `Header.Title` with additional elements like `Breadcrumb` or `Button`, using the sub-components it offers.

## Basic usage

The `Header` component offers the following sub-components to help arrange the items in the page header:
  - The `Header.Title` is the most important sub-component, you must use it with the `Header` component.
  - The `Header.BreadcrumbRow` allows you to display a `Breadcrumb` component or a `Go Back` button above the `Title`.
  - The `Header.Item` allows you to add additional items that will be right-aligned (in desktop views).
  - The `Header.Description` allows you to render some description text below the header's title.

The `Header` component uses its sub-components to arrange the elements you want to insert inside the header. Any child components not wrapped in a sub-component will not be rendered.

```jsx
import * as React from 'react'
import { Header } from '@repay/cactus-web'
...
<Header>
  <Header.Title>Heading Title</Header.Title>
</Header>
```

### Try It out

export const code = `
<Header>
  <Header.BreadcrumbRow>
    <Breadcrumb>
      <Breadcrumb.Item href="/">Home</Breadcrumb.Item>
      <Breadcrumb.Active>Header Docs</Breadcrumb.Active>
    </Breadcrumb>
  </Header.BreadcrumbRow>
  <Header.Title>Heading Title</Header.Title>
  <Header.Description>Description of the current page</Header.Description>
  <Header.Item>
    <Button variant="action">
      Add new configuration
    </Button>
  </Header.Item>
  <Header.Item>Some text because I like crowded UIs</Header.Item>
</Header>
`

<LiveProvider code={code} scope={{ Header, Button, Breadcrumb}}>
  <LiveEditor style={livePreviewStyle} />
  <LiveError />
  <LivePreview />
</LiveProvider>

## Properties

### Header

<PropsTable of={Header} />

### Header.Item

<PropsTable of={Header} staticProp="HeaderItem" />

### Header.Title

<PropsTable of={Header} staticProp="HeaderTitle" />

### Header.BreadcrumbRow

<PropsTable of={Header} staticProp="HeaderBreadcrumbRow" />

### Header.Description
<PropsTable of={Header} staticProp="HeaderDescription" />
