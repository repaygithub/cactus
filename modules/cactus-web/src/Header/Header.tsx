import { mediaGTE } from '@repay/cactus-theme'
import PropTypes from 'prop-types'
import React, { Children, ComponentType, FC, HTMLAttributes, ReactElement } from 'react'
import styled, { css } from 'styled-components'
import { margin, MarginProps } from 'styled-system'

import { withStyles } from '../helpers/styled'
import Text from '../Text/Text'

export type BackgroundColorVariants = 'lightContrast' | 'white'
interface HeaderProps extends MarginProps, HTMLAttributes<HTMLDivElement> {
  bgColor?: BackgroundColorVariants
}

export type HeaderType = FC<HeaderProps> & {
  Item: typeof HeaderItem
  BreadcrumbRow: typeof HeaderBreadcrumbRow
  Title: typeof HeaderTitle
  Description: typeof HeaderDescription
}

export const HeaderItem: FC<HTMLAttributes<HTMLDivElement>> = (props) => <div {...props} />

export const HeaderTitle: FC<React.ComponentProps<typeof Text>> = ({ children, ...props }) => (
  <Text as="h1" textStyle="h2" {...props}>
    {children}
  </Text>
)

export const HeaderBreadcrumbRow: FC<{ children?: React.ReactNode }> = ({ children }) => (
  <>{children}</>
)

export const HeaderDescription: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...rest }) => (
  <StyledDescription {...rest}>{children}</StyledDescription>
)

export const Header: HeaderType = ({ children, bgColor = 'lightContrast', ...rest }) => {
  const childrenArray = Children.toArray(children)
  type ChildElement = ReactElement<any, ComponentType>
  const headerDescription = childrenArray.find(
    (child) => (child as ChildElement).type.displayName === 'HeaderDescription'
  )
  const hasDescription = headerDescription !== undefined

  return (
    <StyledHeader bgColor={bgColor} $hasDescription={hasDescription} {...rest}>
      <MainColumn>
        {childrenArray.find(
          (child) =>
            (child as ChildElement).type.displayName === 'HeaderBreadcrumbRow' && <>{child}</>
        )}
        {childrenArray.find(
          (child) => (child as ChildElement).type.displayName === 'HeaderTitle' && <>{child}</>
        )}
        {headerDescription}
      </MainColumn>
      {childrenArray.some((i) => (i as ChildElement).type.displayName === 'HeaderItem') && (
        <ItemsColumn>
          {childrenArray.filter(
            (child) => (child as ChildElement).type.displayName === 'HeaderItem' && <>{child}</>
          )}
        </ItemsColumn>
      )}
    </StyledHeader>
  )
}

const columnStyles = css`
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
`

const MainColumn = styled.div`
  ${columnStyles}
  ${mediaGTE('small')} {
    padding-top: 0;
    min-width: 40%;
    > div:not(:first-child) {
      margin-left: 0;
    }
    align-items: flex-start;
  }
  flex: 1 1;
  > * {
    max-width: 100%;
    flex-shrink: 0;
  }
`

const ItemsColumn = styled.div`
  ${columnStyles}
  padding-top: 24px;
  > div:not(:first-child) {
    margin-top: 8px;
  }

  ${mediaGTE('small')} {
    align-items: flex-end;
    flex: 1 1 0;
    hyphens: auto;
    overflow-wrap: break-word;
    padding-top: 0;
    max-width: 100%;
    word-wrap: break-word;
    > div {
      margin-top: 0;
      margin-left: 8px;
      word-wrap: break-word;
      max-width: 100%;
      hyphens: auto;
    }
  }

  ${mediaGTE('medium')} {
    flex-direction: row;
    align-items: center;
    flex: 0 1 auto;
    > div:not(:first-child) {
      margin-top: 0;
    }
  }
`

const StyledDescription = styled.div`
  margin-top: ${(p) => `${p.theme.space[3]}px`};
  ${mediaGTE('small')} {
    margin-top: 0;
  }
`

Header.displayName = 'Header'
HeaderBreadcrumbRow.displayName = 'HeaderBreadcrumbRow'
HeaderTitle.displayName = 'HeaderTitle'
HeaderItem.displayName = 'HeaderItem'
HeaderDescription.displayName = 'HeaderDescription'

Header.BreadcrumbRow = HeaderBreadcrumbRow
Header.Title = HeaderTitle
Header.Item = HeaderItem
Header.Description = HeaderDescription

Header.propTypes = {
  bgColor: PropTypes.oneOf(['white', 'lightContrast']),
}

export const StyledHeader = withStyles('header', {
  displayName: 'Header',
  styles: [margin],
  transitiveProps: ['bgColor'],
})<HeaderProps & { $hasDescription: boolean }>`
  align-items: center;
  background-color: ${(p) =>
    p.bgColor ? p.theme.colors[p.bgColor] : p.theme.colors['lightContrast']};
  display: flex;
  flex-direction: column;
  height: auto;
  justify-content: center;
  padding: 8px 16px;
  width: 100%;
  box-sizing: border-box;
  text-align: center;
  word-wrap: break-word;
  overflow-wrap: break-word;
  hyphens: auto;

  ${mediaGTE('small')} {
    text-align: left;
    justify-content: space-between;
    flex-direction: row;
    padding: ${(p) => (p.$hasDescription ? '8px 40px 16px 40px' : '8px 40px')};
  }
`

export default Header
