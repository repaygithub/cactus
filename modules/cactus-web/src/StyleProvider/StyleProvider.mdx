---
name: StyleProvider
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import { livePreviewStyle } from '../helpers/constants'
import StyleProvider from './StyleProvider'
import cactusTheme from '@repay/cactus-theme'

# StyleProvider

The `StyleProvider` is a wrapper for `styled-components`' `ThemeProvider` component which gives you the option to include some global styles.

# Best practices

- Make sure that any components from `@repay/cactus-web` are children of `StyleProvider`. This will ensure that each component has the correct styles.

## Basic usage

```jsx
import React from 'react'
import { StyleProvider } from '@repay/cactus-web'
import cactusTheme from '@repay/cactus-theme'

const App = () => {
  return (
    <StyleProvider theme={cactusTheme} global={true}>
      // Include routes/rest of application here
    </StyleProvider>
  )
}
```

# Breakpoints

The `StyleProvider` has a media queries scale which is based on the provided breaking points and is added to the given theme. The media queries have screen sizes small, medium, large, and extraLarge. The set media queries will be larger than the default, for a mobile first approach.

Props can also vary depending on the current media query. This is done by setting a prop to an array, where the array holds different values for each media query size. The first element of the array maps to the default screen size, and each element after maps to the set media queries from smallest to largest. This is used below in the example code to set the background color.

For more information on Breakpoints and Media Queries visit: https://styled-system.com/theme-specification#breakpoints

## Media Queries Usage

```jsx
import React from 'react'
import { StyleProvider, Box } from '.@repay/cactus-web'
import styled from 'styled-components'

const BreakpointBox = styled(Box)`
  border-radius: 15%;

  ${p => p.theme.mediaQueries.small} {
    border-radius: 25%;
  }
  ${p => p.theme.mediaQueries.medium} {
    border-radius: 35%;
  }
  ${p => p.theme.mediaQueries.large} {
    border-radius: 45%;
  }
  ${p => p.theme.mediaQueries.extraLarge} {
    border-radius: 50%;
  }
`
const backgroundColors = ['springgreen', 'mediumturquoise', 'cornflowerblue', 'thistle', 'pink']

const App = () => {
  return (
    <StyleProvider theme={cactusTheme} global={true}>
      <BreakpointBox backgroundColor={backgroundColors} width="100px" height="100px" />
    </StyleProvider>
  )
}
```

# Properties

<PropsTable of={StyleProvider} staticProp="StyleProvider" />
