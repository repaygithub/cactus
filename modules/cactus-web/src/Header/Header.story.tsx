import { ActionsAdd, ActionsCopy, ActionsDelete } from '@repay/cactus-icons'
import { select, text } from '@storybook/addon-knobs'
import { Meta } from '@storybook/react/types-6-0'
import React from 'react'

import { Breadcrumb, Button, Header, Link, Text } from '../'
import { BackgroundColorVariants } from './Header'

export default {
  title: 'Header',
  component: Header,
} as Meta

const bgColorVariants: BackgroundColorVariants[] = ['lightContrast', 'white']

export const BasicUsage = (): React.ReactElement => {
  const bgSelection = select('Background-color', bgColorVariants, 'lightContrast')
  return (
    <Header bgColor={bgSelection}>
      <Header.Title>{text('Header Title', 'Header')}</Header.Title>
    </Header>
  )
}

export const WithButton = (): React.ReactElement => {
  const bgSelection = select('Background-color', bgColorVariants, 'lightContrast')

  return (
    <Header bgColor={bgSelection}>
      <Header.Title>{text('Header Title', 'Header')}</Header.Title>
      <Header.Item>
        <Button variant="action">
          <ActionsAdd /> Add new configuration
        </Button>
      </Header.Item>
    </Header>
  )
}

export const WithBreadcrumbs = (): React.ReactElement => {
  const bgSelection = select('Background-color', bgColorVariants, 'lightContrast')

  return (
    <Header bgColor={bgSelection}>
      <Header.Title>{text('Header Title', 'Header')}</Header.Title>
      <Header.BreadcrumbRow>
        <Breadcrumb>
          <Breadcrumb.Item href="/">Label</Breadcrumb.Item>
          <Breadcrumb.Active>
            <em>Label</em>
          </Breadcrumb.Active>
        </Breadcrumb>
      </Header.BreadcrumbRow>
      <Header.Item>
        <Button variant="action">
          <ActionsAdd /> Add new configuration
        </Button>
      </Header.Item>
    </Header>
  )
}

export const WithALotOfBreadCrumbs = (): React.ReactElement => {
  const bgSelection = select('Background-color', bgColorVariants, 'lightContrast')

  return (
    <Header bgColor={bgSelection}>
      <Header.Title>{text('Header Title', 'Header')}</Header.Title>
      <Header.BreadcrumbRow>
        <Breadcrumb>
          {[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17].map((i) => (
            <Breadcrumb.Item href="/" key={i}>{`label ${i}`}</Breadcrumb.Item>
          ))}
        </Breadcrumb>
      </Header.BreadcrumbRow>
      <Header.Item>
        <Button variant="action">
          <ActionsAdd /> Add new configuration
        </Button>
      </Header.Item>
      <Header.Item>
        <Button variant="success">
          <ActionsCopy />
        </Button>
      </Header.Item>
      <Header.Item> Some text because I like crowded UIs </Header.Item>
    </Header>
  )
}

export const WithLongTextItem = (): React.ReactElement => {
  const bgSelection = select('Background-color', bgColorVariants, 'lightContrast')

  return (
    <Header bgColor={bgSelection}>
      <Header.BreadcrumbRow>
        <Breadcrumb>
          <Breadcrumb.Item href="/">Label 1</Breadcrumb.Item>
          <Breadcrumb.Item href="/">Label 2</Breadcrumb.Item>
          <Breadcrumb.Item href="/">Label 3</Breadcrumb.Item>
          <Breadcrumb.Item href="/">Label 4</Breadcrumb.Item>
          <Breadcrumb.Active>
            <em>Active label</em>
          </Breadcrumb.Active>
        </Breadcrumb>
      </Header.BreadcrumbRow>
      <Header.Item>
        <Text>You are configuring Merchant directory from OWE Demo Merchant</Text>
      </Header.Item>
      <Header.Title>{text('Header Title', 'Header')}</Header.Title>
    </Header>
  )
}

export const WithMultipleItems = (): React.ReactElement => {
  const bgSelection = select('Background-color', bgColorVariants, 'lightContrast')

  return (
    <Header bgColor={bgSelection}>
      <Header.BreadcrumbRow>
        <Breadcrumb>
          <Breadcrumb.Item href="/">Label</Breadcrumb.Item>
          <Breadcrumb.Active>
            <em>Label</em>
          </Breadcrumb.Active>
        </Breadcrumb>
      </Header.BreadcrumbRow>
      <Header.Item>
        <Button variant="action">
          <ActionsAdd /> Add new configuration
        </Button>
      </Header.Item>
      <Header.Item>
        <Button variant="success">
          <ActionsCopy />
        </Button>
      </Header.Item>
      <Header.Item> Some text because I like crowded UIs </Header.Item>
      <Header.Title>{text('Header Title', 'Header')}</Header.Title>
    </Header>
  )
}

export const WithGoBackLink = (): React.ReactElement => {
  const bgSelection = select('Background-color', bgColorVariants, 'lightContrast')

  return (
    <Header bgColor={bgSelection}>
      <Header.BreadcrumbRow>
        <Link to="https://repaygithub.github.io/cactus/">Go back!</Link>
      </Header.BreadcrumbRow>
      <Header.Title>{text('Header Title', 'Header')}</Header.Title>
      <Header.Item>
        <Button variant="action">
          <ActionsAdd /> Add new configuration
        </Button>
      </Header.Item>
    </Header>
  )
}

export const WithDescription = (): React.ReactElement => {
  const descriptionText = text('Description Text', 'I am describing something about this page')
  return (
    <Header>
      <Header.Title>I Have a Description</Header.Title>
      <Header.Description text={descriptionText} />
    </Header>
  )
}

export const WithEverything = (): React.ReactElement => {
  const descriptionText = text(
    'Description Text',
    'This story was put here to showcase the header with all sub-components that it offers'
  )
  return (
    <Header>
      <Header.BreadcrumbRow>
        <Breadcrumb>
          <Breadcrumb.Item href="/">Home</Breadcrumb.Item>
          <Breadcrumb.Active>Full Header</Breadcrumb.Active>
        </Breadcrumb>
      </Header.BreadcrumbRow>
      <Header.Title>I Am a Jam-Packed Header</Header.Title>
      <Header.Description>{descriptionText}</Header.Description>
      <Header.Item>
        <Button variant="action">
          <ActionsAdd /> Add new configuration
        </Button>
      </Header.Item>
      <Header.Item>
        <Button variant="success">
          <ActionsCopy />
        </Button>
      </Header.Item>
      <Header.Item>
        <Button variant="danger">
          <ActionsDelete />
        </Button>
      </Header.Item>
    </Header>
  )
}
