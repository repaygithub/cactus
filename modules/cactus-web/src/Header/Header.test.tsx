import { render } from '@testing-library/react'
import * as React from 'react'

import { StyleProvider } from '../StyleProvider/StyleProvider'
import Header from './Header'

describe('component: Header', () => {
  test('Header component should change their background color by passing the bgColor prop', () => {
    const { getByTestId } = render(
      <StyleProvider>
        <Header bgColor="white" data-testid="headerComponent"></Header>
      </StyleProvider>
    )

    const rawHeader = getByTestId('headerComponent')
    const headerStyles = window.getComputedStyle(rawHeader)
    expect(headerStyles.backgroundColor).toBe('rgb(255, 255, 255)')
  })
})
