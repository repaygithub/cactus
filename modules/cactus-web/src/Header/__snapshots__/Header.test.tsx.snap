// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Header Should render the Header with Title and Items 1`] = `
.c4 {
  position: relative;
  padding: 2px 30px;
  outline: none;
  cursor: pointer;
  overflow: visible;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.5;
  border: 1px solid;
  border-radius: 8px;
  color: hsl(200,96%,11%);
  background-color: hsl(0,0%,100%);
  border-color: hsl(200,96%,11%);
}

.c4::-moz-focus-inner {
  border: 0;
}

.c4:focus::after {
  content: '';
  display: block;
  position: absolute;
  height: calc(100% + 10px);
  width: calc(100% + 10px);
  top: -5px;
  left: -5px;
  border: 1px solid;
  border-radius: 8px;
  border-color: hsl(200,96%,35%);
  box-sizing: border-box;
}

.c4 svg {
  display: inline-block;
  margin-top: -4px;
}

.c4 .sc-gyUeRy {
  position: absolute;
  left: 50%;
  top: 50%;
  margin-left: -8px;
  margin-top: -8px;
}

.c4:hover {
  background-color: hsl(200,96%,11%);
  color: hsl(0,0%,100%);
  border-color: hsl(200,96%,11%);
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 24px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(200,29%,90%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 0;
  width: 100%;
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 18px;
    line-height: 1.5;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding-top: 0px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c1:last-child {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    padding-top: 0px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c3:last-child {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 8px 40px;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        Heading Title
      </h2>
    </div>
    <div
      class="c3"
    >
      <button
        aria-live="assertive"
        class="c4"
        type="button"
      >
        <span>
          Click me!
        </span>
      </button>
      I'll be back
    </div>
  </header>
</div>
`;

exports[`component: Header Should render the Header with Title and Items 2`] = `
.c4 {
  position: relative;
  padding: 2px 30px;
  outline: none;
  cursor: pointer;
  overflow: visible;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.5;
  border: 1px solid;
  border-radius: 8px;
  color: hsl(200,96%,11%);
  background-color: hsl(0,0%,100%);
  border-color: hsl(200,96%,11%);
}

.c4::-moz-focus-inner {
  border: 0;
}

.c4:focus::after {
  content: '';
  display: block;
  position: absolute;
  height: calc(100% + 10px);
  width: calc(100% + 10px);
  top: -5px;
  left: -5px;
  border: 1px solid;
  border-radius: 8px;
  border-color: hsl(200,96%,35%);
  box-sizing: border-box;
}

.c4 svg {
  display: inline-block;
  margin-top: -4px;
}

.c4 .sc-gyUeRy {
  position: absolute;
  left: 50%;
  top: 50%;
  margin-left: -8px;
  margin-top: -8px;
}

.c4:hover {
  background-color: hsl(200,96%,11%);
  color: hsl(0,0%,100%);
  border-color: hsl(200,96%,11%);
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 24px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(200,29%,90%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 0;
  width: 100%;
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 18px;
    line-height: 1.5;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding-top: 0px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c1:last-child {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    padding-top: 0px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c3:last-child {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 8px 40px;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        Hasta la vista, baby
      </h2>
    </div>
    <div
      class="c3"
    >
      <button
        aria-live="assertive"
        class="c4"
        type="button"
      >
        <span>
          Click me!
        </span>
      </button>
      I'll be back
    </div>
  </header>
</div>
`;

exports[`component: Header Should render the Header with Title, Breadcrumb and Items 1`] = `
.c4 {
  font-size: 8px;
  vertical-align: middle;
}

.c3 {
  color: black;
  font-style: normal;
  outline: none;
}

.c3:visited,
.c3:link {
  color: hsl(200,9%,35%);
  font-style: normal;
}

.c3:hover {
  color: hsl(200,96%,35%);
}

.c3:focus {
  outline: hsl(200,96%,35%) solid 1px;
}

.c5 {
  color: hsl(200,18%,45%);
  margin: 0 3px;
  font-size: 10px;
}

.c2 {
  font-size: 15px;
  line-height: 1.6;
}

.c2 > ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  list-style: none;
  padding: 0;
  margin: 0;
}

.c2 [aria-current='page'] {
  color: hsl(200,10%,20%);
  font-style: italic;
}

.c8 {
  position: relative;
  padding: 2px 30px;
  outline: none;
  cursor: pointer;
  overflow: visible;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.5;
  border: 1px solid;
  border-radius: 8px;
  color: hsl(200,96%,11%);
  background-color: hsl(0,0%,100%);
  border-color: hsl(200,96%,11%);
}

.c8::-moz-focus-inner {
  border: 0;
}

.c8:focus::after {
  content: '';
  display: block;
  position: absolute;
  height: calc(100% + 10px);
  width: calc(100% + 10px);
  top: -5px;
  left: -5px;
  border: 1px solid;
  border-radius: 8px;
  border-color: hsl(200,96%,35%);
  box-sizing: border-box;
}

.c8 svg {
  display: inline-block;
  margin-top: -4px;
}

.c8 .sc-gyUeRy {
  position: absolute;
  left: 50%;
  top: 50%;
  margin-left: -8px;
  margin-top: -8px;
}

.c8:hover {
  background-color: hsl(200,96%,11%);
  color: hsl(0,0%,100%);
  border-color: hsl(200,96%,11%);
}

.c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 24px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(200,29%,90%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 0;
  width: 100%;
}

@media screen and (min-width:1024px) {
  .c2 {
    font-size: 15px;
    line-height: 1.6;
  }
}

@media screen and (min-width:1024px) {
  .c8 {
    font-size: 18px;
    line-height: 1.5;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding-top: 0px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c1:last-child {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    padding-top: 0px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c7:last-child {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 8px 40px;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <nav
        aria-label="Breadcrumb"
        class="c2"
      >
        <ul>
          <li>
            <a
              class="c3"
              href="/"
            >
              Label
            </a>
            <svg
              class="c4 c5"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.7071068,2.70710678 C18.0976311,2.31658249 18.0976311,1.68341751 17.7071068,1.29289322 C17.3165825,0.902368927 16.6834175,0.902368927 16.2928932,1.29289322 L6.29289322,11.2928932 C5.90236893,11.6834175 5.90236893,12.3165825 6.29289322,12.7071068 L16.2928932,22.7071068 C16.6834175,23.0976311 17.3165825,23.0976311 17.7071068,22.7071068 C18.0976311,22.3165825 18.0976311,21.6834175 17.7071068,21.2928932 L8.41421356,12 L17.7071068,2.70710678 Z"
                transform="matrix(-1 0 0 1 24 0)"
              />
            </svg>
          </li>
          <div
            aria-current="page"
          >
            <em>
              Label
            </em>
          </div>
        </ul>
      </nav>
      <h2
        class="c6"
      >
        Heading Title
      </h2>
    </div>
    <div
      class="c7"
    >
      <button
        aria-live="assertive"
        class="c8"
        type="button"
      >
        <span>
          Click me!
        </span>
      </button>
      I'll be back
    </div>
  </header>
</div>
`;

exports[`component: Header Should render the Header witha a single title 1`] = `
.c2 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(200,29%,90%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 0;
  width: 100%;
}

@media screen and (min-width:768px) {
  .c1 {
    padding-top: 0px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c1:last-child {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 8px 40px;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        Hasta la vista, baby
      </h2>
    </div>
  </header>
</div>
`;
